skypy/power_spectrum/_halofit.py

```diff
@@ -98,21 +105,32 @@ def halofit(wavenumber, redshift, linear_power_spectrum,
 
     Examples
     --------
+
+    This will return the halofit power spectrum for a given wavenumber,
+    redshift and linear power spectrum:
+
     >>> import numpy as np
     >>> from astropy.cosmology import FlatLambdaCDM
-    >>> kvec = np.array([1.00000000e-04, 1.01000000e+01])
+    >>> cosmo = FlatLambdaCDM(H0=67.04, Om0=0.21479, Ob0=0.04895)
+
+    The Takahashi halofit power spectrum in :math:`Mpc^3` at redshift 0 and at
+    wavenumers :math:`k=0.0001/Mpc` and :math:`k=10.1/Mpc`:
+
     >>> zvalue = 0.0
+    >>> kvec = np.array([1.00000000e-04, 1.01000000e+01])
     >>> pvec = np.array([388.6725682632502, 0.21676249605280398])
-    >>> cosmo = FlatLambdaCDM(H0=67.04, Om0=0.21479, Ob0=0.04895)
-    >>> halofit(kvec, zvalue, pvec, cosmo, _takahashi_parameters)
+    >>> halofit_takahashi(kvec, zvalue, pvec, cosmo)
     array([388.67064424,   0.72797614])
+
     '''
 
     # Manage shapes of input arrays
     return_shape = np.shape(linear_power_spectrum)
     redshift = np.atleast_1d(redshift)
+
     if np.ndim(linear_power_spectrum) == 1:
-        linear_power_spectrum = linear_power_spectrum[:, np.newaxis]
+        length_pk = len(linear_power_spectrum)
+        linear_power_spectrum = linear_power_spectrum.reshape(1, length_pk)
```

Comment:
```suggestion
        linear_power_spectrum = linear_power_spectrum.reshape(1, -1)
```

Reply:
Why this change?

Reply:
it seems rather pointless to get the length explicitly just to add a dimension -- alternatives are having a `None` index, or using `expand_dims`, but I think the implicit reshape explains best what happens

Reply:
makes sense! Thanks